src/NodaTime.Test/Text/TypeConvertersTest.cs

```diff
@@ -116,18 +118,30 @@ public void HasConverter(Type type)
         public void Period_Roundtrip(int years, int months, int weeks, int days, long hours, long minutes, long seconds, long milliseconds, long ticks, long nanoseconds, string text) =>
             AssertRoundtrip(text, new Period(years, months, weeks, days, hours, minutes, seconds, milliseconds, ticks, nanoseconds));
 
-        private static void AssertRoundtrip<T>(string input, T expected)
+        [Test]
+        [TestCase(2019, 1, 1, 4, 59, "Europe/Paris", "2019-01-01T04:59:00 Europe/Paris (+01)")]
+        [TestCase(2020, 2, 29, 23, 59, "America/Los_Angeles", "2020-02-29T23:59:00 America/Los_Angeles (-08)")]
+        [TestCase(2018, 12, 31, 13, 30, "Asia/Kathmandu", "2018-12-31T13:30:00 Asia/Kathmandu (+05:45)")]
+        public void ZonedDateTime_Roundtrip(int year, int month, int day, int hour, int minute, string zoneId, string text)
+        {
+            Assert.AreSame(DateTimeZoneProviders.Tzdb, DateTimeZoneProviders.ForTypeConverter,
```

Comment:
Is this better than setting it explicitly in the test?

(Tests are run on a single thread, I assume?)

Reply:
They're run on a single thread, but the concern is really that if we ever introduce another test that changes the `ForTypeConverter` property and fetches a type converter, it'll break this test without us understanding why. Just setting the property doesn't necessarily help, because of the caching.